packages/rxsocketio/src/operators/send.js

```diff
@@ -11,24 +31,75 @@ function getMessageArgs(message) {
   return [topic, formattedMessage, ...binary];
 }
 
+function sendWithTransactions({
+  message,
+  socket,
+  action,
+  // If true, messages will wait in line for the prior message to be received
+  // before the next is emitted. (It forces sequential data transfer.)
+  maxTimeout,
+  // Should take a response and return true if response is valid, otherwise
+  // false:
+  validateResponse,
+  shouldRetry,
+  retryDelay,
+  retryCount,
+  retryFunction, // custom function to pass to retryWhen operator
+}) {
+  const transaction$ = new Observable(obs => {
+    socket
+      .binary(!!message.binary)
+      .compress(!!message.binary)
+      .emit(...getMessageArgs(message), (...responseArgs) => {
+        if (!validateResponse(responseArgs)) {
+          return obs.error(new Error('Server did not acknowledge response'));
+        }
+        obs.next([socket, action]);
+        return obs.complete();
+      });
+  });
+  return transaction$.pipe(
+    timeout(maxTimeout),
+    (
+      shouldRetry
+      ? retryWhen(retryFunction || (
+          shouldRetry
+          ? error$ => error$.pipe(delay(retryDelay), take(retryCount))
+          : () => of()
+      ))
+      : tap(() => 1)
+    )
+  );
+}
```

Comment:
this logic is kind of neat.  basically, if the the conduit is operating in `useTransactions` mode, it will wait until the server confirms receipt of a message and can have some retry logic built in in case the message doesn't send properly.